From b474469efdd4a47dc38ba063311c23097fe2332f Mon Sep 17 00:00:00 2001
From: Sujeet Baranwal <sbaranwal@marvell.com>
Date: Tue, 29 Sep 2020 10:30:16 -0700
Subject: [PATCH 1456/1921] comphy: appropriate base address

The base address for comphy is different than
used here. In order to swich on/off comphy, smc shall
be called with appropriate address.

Change-Id: Icb791bb90a55e7ed4cec9b316a83246757c9ea34
Signed-off-by: Sujeet Baranwal <sbaranwal@marvell.com>
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/50165
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-cp11x.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
index cf8be27e128c..99619d9f17cb 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
@@ -136,9 +136,9 @@
 			reg-names = "pp", "mspg", "cm3";
 		};
 
-		CP11X_LABEL(comphy): phy@120000 {
+		CP11X_LABEL(comphy): phy@441000 {
 			compatible = "marvell,comphy-cp110";
-			reg = <0x120000 0x6000>;
+			reg = <0x441000 0xf4>, <0x120000 0x6000>;
 			marvell,system-controller = <&CP11X_LABEL(syscon0)>;
 			clocks = <&CP11X_LABEL(clk) 1 5>, <&CP11X_LABEL(clk) 1 6>,
 				 <&CP11X_LABEL(clk) 1 18>;
-- 
2.31.1

